rest/route/version.go

```diff
@@ -63,6 +64,59 @@ func (vh *versionHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(versionModel)
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/versions/{version_id}
+
+type versionPatchHandler struct {
+	Activated *bool `json:"activated"`
+
+	versionId string
+}
+
+func makePatchVersion() gimlet.RouteHandler {
+	return &versionPatchHandler{}
+}
+
+func (vh *versionPatchHandler) Factory() gimlet.RouteHandler {
+	return &versionPatchHandler{}
+}
+
+// Parse fetches the versionId from the http request.
+func (vh *versionPatchHandler) Parse(ctx context.Context, r *http.Request) error {
+	if err := utility.ReadJSON(r.Body, vh); err != nil {
+		return errors.Wrap(err, "reading body")
+	}
+	if vh.Activated == nil {
+		return gimlet.ErrorResponse{
+			Message:    "Must set 'activated'",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	vh.versionId = gimlet.GetVars(r)["version_id"]
+	if vh.versionId == "" {
+		return errors.New("request data incomplete")
```

Comment:
should we specify that it is missing a versionId?

Reply:
do we want to elaborate on this error message?

Reply:
sorry i didn't see this before!